src/main/java/com/linkedin/kmf/services/CommitLatencyMetrics.java

```diff
@@ -30,25 +30,76 @@
   private static final String METRIC_GROUP_NAME = "commit-latency-service";
   private static final Logger LOG = LoggerFactory.getLogger(CommitLatencyMetrics.class);
   private final Sensor _commitOffsetLatency;
+  private long _commitStartTimeMs;
+  private static boolean inProgressCommit;
```

Comment:
3 things.

1. Why does the variable name not start with `_` (`_ inProgressCommit` instead of `inProgressCommit`)

2. Why is this variable defined as `static`?

3. I think this variable might need to be defined as a volatile boolean variable. Since `recordCommitStart` and `recordCommitComplete` are called in different threads (IIUC). `recordCommitStart` is called in the `consume()` method where as `recordCommitComplete` is called in a callback.

Maybe you already know this. But in general, if multiple threads could change and read the state of some shared variable, that shared variable needs to be defined as volatile.
See https://www.logicbig.com/tutorials/core-java-tutorial/java-multi-threading/visibility-volatile.html

Reply:
1 - Because `inProgressCommit` is a private static boolean variable and not a private non-static boolean variable.
2 - Let's use non-static since we don't want to provide class level access to a this variable.
3 - sgtm